x-pack/test/security_functional/tests/expired_session/basic_functionality.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { SESSION_ERROR_REASON_HEADER } from '@kbn/security-plugin/common/constants';
+import { setTimeout as setTimeoutAsync } from 'timers/promises';
+import { parse } from 'url';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const browser = getService('browser');
+  const deployment = getService('deployment');
+  const PageObjects = getPageObjects(['security', 'common']);
+
+  describe('Basic functionality', function () {
+    this.tags('includeFirefox');
+
+    before(async () => {
+      await PageObjects.security.forceLogout();
+    });
+
+    afterEach(async () => {
+      // NOTE: Logout needs to happen before anything else to avoid flaky behavior
+      await PageObjects.security.forceLogout();
+    });
+
+    it('should attach msg=SESSION_EXPIRED to the redirect URL when redirecting to /login if the session has expired when trying to access a page', async () => {
```

Comment:
Would there be any benefit to testing the other cases (e.g. AUTHENTICATION_ERROR)? Or are they not relevant IRL?

Reply:
My idea was that the new test suite defined in `x-pack/test/security_functional/expired_session.config.ts` should only concern itself with session expiration, which is the topic of this PR. I didn't check to see if there were already functional tests for `AUTHENTICATION_ERROR`, but if there's not, then I agree we should create them. I would prefer it to not block this PR however if it turns out to be complicated to add (I haven't checked).

Reply:
It looks like the redirect URL msg is tested for both the session expired and auth error cases in session_expired.test.ts